src/entities/options/exercisable-option.ts

```diff
@@ -0,0 +1,44 @@
+import type { PublicClient } from '@wagmi/core';
+import type { Trader } from '../trader/base-trader';
+import type { SimulatedTxRequest } from '../../types';
+import { OptionType } from './option-type';
+
+export class Option extends OptionType {
+  constructor({
+    optionId,
+    publicClient,
+  }: {
+    optionId: bigint;
+    publicClient: PublicClient;
+  }) {
+    super({ tokenId: optionId, publicClient });
+  }
+
+  public async exerciseOption({
+    amount,
+    trader,
+  }: {
+    amount: bigint;
+    trader: Trader;
+  }) {
+    if (!this.tokenId) {
+      console.log('Missing TokenId');
+      return undefined;
+    }
+    // prepare tx
+    const { request } = await this.clearinghouse.simulate.exercise([
+      this.tokenId,
+      amount,
+    ]);
```

Comment:
This will simulate the call not from the `trader`'s wallet, but rather from a `publicClient` that was passed to the `constructor`.

What this possibly could lead to is passing a `trader` instance that has a different `walletClient` property rather than the one passed to the `this.clearinghouse`, hence passing the simulation but failing the transaction execution.

Reply:
Good catch, thank you